atomate/vasp/workflows/base/hubbard_hund_linresp.py

```diff
@@ -0,0 +1,639 @@
+# coding: utf-8
+
+import os
+import itertools
+
+from atomate.vasp.fireworks.core import StaticFW
+from fireworks import Workflow, Firework
+from atomate.vasp.powerups import (
+    add_tags,
+    add_additional_fields_to_taskdocs,
+    add_wf_metadata,
+    add_common_powerups,
+)
+from atomate.vasp.workflows.base.core import get_wf
+
+from atomate.vasp.firetasks.run_calc import RunVaspCustodian
+from atomate.common.firetasks.glue_tasks import PassCalcLocs
+from atomate.vasp.firetasks.glue_tasks import CopyVaspOutputs
+from atomate.vasp.firetasks.parse_outputs import (
+    VaspToDb, HubbardHundLinRespToDb
+)
+from atomate.vasp.firetasks.write_inputs import WriteVaspFromIOSet
+
+from atomate.utils.utils import get_logger
+
+logger = get_logger(__name__)
+
+from atomate.vasp.config import VASP_CMD, DB_FILE, ADD_WF_METADATA
+
+from atomate.vasp.workflows.presets.scan import wf_scan_opt
+from uuid import uuid4
+from pymatgen.io.vasp.sets import MPRelaxSet, MPStaticSet
+from pymatgen.io.vasp.inputs import Poscar, Incar
+from pymatgen.core import Lattice, Structure, Element
+
+import numpy as np
+
+__author__ = "Guy Moore, Martin Siron"
+__maintainer__ = "Guy Moore"
+__email__ = "gmoore@lbl.gov"
+__status__ = "Production"
+__date__ = "February 2020"
+
+__hubbard_hund_linresp_wf_version__ = 0.0
+
+module_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)))
+
+
+def get_wf_hubbard_hund_linresp(
+        structure, uis,
+        relax_nonmagnetic=True,
+        spin_polarized=True,
+        applied_potential_range=[-0.2, 0.2],
+        num_parallel_evals=9, num_total_evals=9,
+        site_indices_u=None, species_u=None,
+        find_nearest_sites=True,
+        parallel_scheme=1,
+        ediff_tight=1.0e-6,
+        c=None):
+    """
+    Compute Hubbard U (and Hund J) on-site interaction values using GGA+U
+    linear response method proposed by Cococcioni et. al.
+    (DOI: 10.1103/PhysRevB.71.035105)
+    and the spin-polarized response formalism developed by Linscott et. al.
+    (DOI: 10.1103/PhysRevB.98.235157)
+
+    Args:
+        structure:
+        spin_polarized: Perform spin-dependent perturbations
+        applied_potential_range: Bounds of applied potential
+        num_evals: Number of perturbation evalutaions
+        uis: user INCAR settings (must set LMAXMIX, i.e. uis = {"LMAXMIX": 4})
+        site_indices_u: List of site indices within
+    Structure indicating perturbation sites
+        species_u: List of names of species (string)
+    of sites to perturb; First site of that species
+    is selected in the structure
+        use_default_uvals: Use the default U values
```

Comment:
`use_default_uvals` could perhaps be more general, e.g. so user can provide specific U values to use for other sites (this would just get passed into `user_incar_settings` ultimately, so perhaps this isn't necessary if the user can set the default uvals via `uis`)